src/latexify/codegen/identifier_converter.py

```diff
@@ -41,14 +51,51 @@ def convert(self, name: str) -> tuple[str, bool]:
                 - latex: Corresponding LaTeX expression.
                 - is_single_character: Whether `latex` can be treated as a single
                     character or not.
+        Raises:
+            LatexifyError: Resulting latex is not valid. This most likely occurs where the
+            symbol starts or ends with an underscore, and escape_underscores=False.
         """
+        if not self._escape_underscores and "_" in name:
+            # Check if we are going to generate an invalid Latex string. Better to raise an
+            # exception here than have the resulting Latex fail to compile/display
+            if "__" in name:
+                raise ValueError(
+                    f'Identifier "{name}" has a double underscore will result in '
+                    "invalid Latex when underscores are not escaped"
+                )
+            if name.startswith("_"):
+                raise ValueError(
+                    f'Identifier "{name}" starts with an underscore will result in '
+                    "invalid Latex when underscores are not escaped"
+                )
+            if name.endswith("_"):
+                raise ValueError(
+                    f'Identifier "{name}" ends with an underscore will result in '
+                    "invalid Latex when underscores are not escaped"
+                )
+            elems = [
+                IdentifierConverter(
+                    use_math_symbols=self._use_math_symbols,
+                    use_mathrm=False,
+                    escape_underscores=True,
+                ).convert(n)[0]
```

Comment:
```suggestion
                self.convert(n)[0]
```

Reply:
Can't use `self` here as we need to ensure `use_mathrm` is `False`, we only want mathrm{} to be wrapped around the whole identifier, not each sub-identifier. `escape_underscores=True` was added to prevent an infinite recursion and to ensure line 98 triggers correctly

Reply:
ah okay my bad. thanks for clarifying.